app/src/features/apiClient/screens/apiClient/components/sidebar/collectionsList/CollectionsList.tsx

```diff
@@ -1,18 +1,60 @@
-import React from "react";
+import React, { useCallback, useMemo } from "react";
 import { RQAPI } from "features/apiClient/types";
 import { Typography } from "antd";
-import placeholderImage from "../../../../../../../assets/images/illustrations/empty-sheets-dark.svg";
-import { NavLink } from "react-router-dom";
-import PATHS from "config/constants/sub/paths";
 import { useApiClientContext } from "features/apiClient/contexts";
-import { REQUEST_METHOD_COLORS } from "../../../../../../../constants";
+import { NewRecordNameInput } from "./newRecordNameInput/NewRecordNameInput";
+import { CollectionRow } from "./collectionRow/CollectionRow";
+import { RequestRow } from "./requestRow/RequestRow";
+import { isApiCollection, isApiRequest } from "../../../utils";
+import { ApiRecordEmptyState } from "./apiRecordEmptyState/ApiRecordEmptyState";
 import "./collectionsList.scss";
 
-interface Props {}
+interface Props {
+  onNewClick: (src: RQAPI.AnalyticsEventSource) => void;
+  recordTypeToBeCreated: RQAPI.RecordType;
+  isNewRecordNameInputVisible: boolean;
+  hideNewRecordNameInput: () => void;
+}
 
-export const CollectionsList: React.FC<Props> = () => {
+export const CollectionsList: React.FC<Props> = ({
+  onNewClick,
+  recordTypeToBeCreated,
+  isNewRecordNameInputVisible,
+  hideNewRecordNameInput,
+}) => {
   const { isLoadingApiClientRecords, apiClientRecords } = useApiClientContext();
 
+  const prepareRecordsToRender = useCallback((records: RQAPI.Record[]) => {
+    const collections: Record<RQAPI.CollectionRecord["id"], RQAPI.CollectionRecord> = {};
+    const requests: RQAPI.ApiRecord[] = [];
+
+    records.forEach((record) => {
+      if (isApiCollection(record)) {
+        collections[record.id] = { ...record, data: { ...record.data, children: [] } };
+      }
+    });
+
+    records.forEach((record) => {
+      if (isApiRequest(record)) {
+        if (record.collectionId) {
+          collections[record.collectionId].data.children.push(record);
+        } else {
+          requests.push(record);
+        }
+      }
+    });
```

Comment:
This can be improved, currently it is iterating the `records` twice.

Reply:
```js
records.forEach((record) => {
    if (isApiCollection(record)) {
      collections[record.id] = {
        ...record,
        data: {
          ...record.data,
          children: collections[record.id]?.data.children || []
        }
      };
    } else if (isApiRequest(record)) {
      if (record.collectionId) {
        if (!collections[record.collectionId]) {
          collections[record.collectionId] = {
           // create empty collection
          }
        }
        collections[record.collectionId].data.children.push(record);
      } else {
        requests.push(record);
      }
    }
  });
```

Reply:
Added a `TODO` comment for now